test/rulesets/Base/mapreduce.jl

```diff
@@ -203,4 +267,60 @@ end
             @test unthunk(back(fill(0.5, 2, 2))[2]) ≈ [1/2 0; 0 0]  # ProjectTo'd to Diagonal now
         end
     end
+    @testset "accumulate(f, ::Array)" begin
+        CFG = ChainRulesTestUtils.ADviaRuleConfig()
+
+        # Simple
+        y1, b1 = rrule(CFG, accumulate, *, [1,2,3,4]; init=1)
+        @test y1 == [1, 2, 6, 24]
+        @test b1([1,1,1,1]) == (NoTangent(), NoTangent(), [33, 16, 10, 6])
+
+        y2, b2 = rrule(CFG, accumulate, /, [1 2; 3 4])
+        @test y2 ≈ accumulate(/, [1 2; 3 4])
+        @test b2(ones(2,2))[3] ≈ [1.5416666 -0.104166664; -0.18055555 -0.010416667]  atol=1e-6
+        # ForwardDiff.gradient(x -> sum(accumulate(/,x)), Float32[1 2; 3 4]) |> string
+
+        # Test execution order
+        c3 = Counter()
+        y3, b3 = rrule(CFG, accumulate, c3, [5,7,11]; init=3)
+        @test c3 == Counter(3)
+        @test y3 == [8, 30, 123] == accumulate(Counter(), [5,7,11]; init=3)
+        @test b3([1,1,1]) == (NoTangent(), NoTangent(), [29169, 602, 23]) # the 23 is clear!
+
+        c4 = Counter()
+        y4, b4 = rrule(CFG, accumulate, c4, [5,7,11])
+        @test c4 == Counter(2)
+        @test y4 == [5, (5+7)*1, ((5+7)*1 + 11)*2] == accumulate(Counter(), [5,7,11])
```

Comment:
Do you want to run the BlueStyle formatter on this block of changed code?
Spacing is off.